diff --git a/target/linux/ipq807x/base-files/etc/init.d/bootcount b/target/linux/ipq807x/base-files/etc/init.d/bootcount
index 6917446a9b..a38432c387 100755
--- a/target/linux/ipq807x/base-files/etc/init.d/bootcount
+++ b/target/linux/ipq807x/base-files/etc/init.d/bootcount
@@ -10,4 +10,9 @@ boot() {
 		fw_setenv changed
 	;;
 	esac
+
+	for q in $(ls /sys/class/net/eth*/queues/rx-*/rps_cpus); do echo f > $q; done
+	for q in $(ls /sys/class/net/eth*/queues/rx-*/rps_flow_cnt); do echo 4096 > $q; done
+	for q in $(ls /sys/class/net/eth*/queues/tx-*/xps_cpus); do echo f > $q; done
+	echo 32768 > /proc/sys/net/core/rps_sock_flow_entries
 }
